floor_generator/lib/writer/query_method_writer.dart

```diff
@@ -94,7 +94,11 @@ class QueryMethodWriter implements Writer {
     return _queryMethod.parameters
         .map((parameter) {
           if (!parameter.type.isDartCoreList) {
-            return parameter.displayName;
+            if (parameter.type.isDartCoreBool) {
+              return '${parameter.displayName} == null ? null : ${parameter.displayName} ? 1 : 0';
```

Comment:
⚠️This mapping also is quite advanced and not very readable. Let's talk about how we cold simplify that.

Please add some test for this new functionality.

Reply:
Added tests and parentheses like already mentioned.